tests/php/Tasks/InvalidateAllSessionsTaskTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace SilverStripe\SessionManager\Tests\Services;
+
+use SilverStripe\Control\Middleware\ConfirmationMiddleware\Url;
+use SilverStripe\Control\Tests\HttpRequestMockBuilder;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\Security\AuthenticationHandler;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\RememberLoginHash;
+use SilverStripe\Security\Security;
+use SilverStripe\SessionManager\Extensions\RememberLoginHashExtension;
+use SilverStripe\SessionManager\Middleware\LoginSessionMiddleware;
+use SilverStripe\SessionManager\Models\LoginSession;
+use SilverStripe\SessionManager\Tasks\InvalidateAllSessionsTask;
+
+class InvalidateAllSessionsTaskTest extends SapphireTest
+{
+    use HttpRequestMockBuilder;
+
+    protected static $fixture_file = 'InvalidateAllSessionsTaskTest.yml';
+
+    protected static $required_extensions = [
+        RememberLoginHash::class => [
+            RememberLoginHashExtension::class,
+        ],
+    ];
+
+    public function testRemoveAllSessions()
+    {
+        // Ensure existing fixtured sessions are valid
+        DBDatetime::set_mock_now('2003-02-15 10:00:00');
+
+        // Log in as member 1 and make them active
+        $request = $this->buildRequestMock('dev/tasks/' . InvalidateAllSessionsTask::config()->get('segment'));
+        $request->method('getIP')->willReturn('192.168.0.1');
+        $member = $this->objFromFixture(Member::class, 'member1');
+        $handler = Injector::inst()->get(AuthenticationHandler::class);
+        $handler->login($member, true, $request);
+
+        // Ensure fixtured sessions are created and at least one login hash is generated
+        $this->assertNotSame(0, LoginSession::get()->count(), 'There are no LogginSessions');
+        $this->assertNotSame(0, RememberLoginHash::get()->count(), 'There are no RememberLoginHashes');
+
+        // Completely invalidate all current sessions and login hashes
+        $task = new InvalidateAllSessionsTask();
+        $task->run($request);
+
+        // Assert all sessions are removed from the database
+        $this->assertSame(0, LoginSession::get()->count(), 'There is at least one LoginSession');
+        $this->assertSame(0, RememberLoginHash::get()->count(), 'There is at least one RememberLoginHash');
+
+        // Navigate to re-validate current session
+        $middleware = new LoginSessionMiddleware(new Url('no-match'));
+        $next = false;
+        $middleware->process(
+            $request,
+            function () use (&$next) {
+                $next = true;
+            }
+        );
+        // Assert user who initiated the request's session was also invalidated
+        $this->assertNull(
+            Security::getCurrentUser(),
+            'user was logged out because their session was invalidated'
```

Comment:
```suggestion
            'User was not logged out when their session was invalidated'
```

Reply:
Ahh, I must have misunderstood the messages - I thought it was intended to say "this is what went wrong" but it seems that it instead is meant to be "this is what we are testing"

Reply:
The way tests in this module are written, the comments are "this is what this assertion is testing" as opposed to "this is what went wrong". I was trying to be consistent with the other tests in this module... should I update the other tests in this module to also have a message indicating what went wrong, as opposed to what is being tested?

Reply:
er, the assertions directly above, the message are about saying what went wrong

`$this->assertNotSame(0, LoginSession::get()->count(), 'There are no LoginSessions');` aka
`$this->assertSame(1, LoginSession::get()->count(), 'There are no LoginSessions');`

Reply:
Yup, good point. I'm getting my wires crossed here.
I had a look and phpunit don't opine on what the message should be used for, so I've amended those to be consistent with this one, which in turn is consistent with the messages in other tests which are already in this module.